src/package-scanner.ts

```diff
@@ -0,0 +1,192 @@
+    for (const document of this.getPackageDocuments()) {
+      this.scanDocument(document);
+    }
+
+    return this.getResults();
+  }
+
+  /**
+   * Get all relevant HTML documents from a package that should be scanned,
+   * coverted, or otherwise handled by the modulizer.
+   */
+  getPackageDocuments() {
+    return [
+      ...this.analysis.getFeatures(
+          {kind: 'html-document', externalPackages: true})
+    ].filter((d) => {
+      // Filter out any inline documents returned by the analyzer
+      if (d.isInline === true) {
+        return false;
+      }
+      // Filter out any excluded documents
+      const documentUrl = this.urlHandler.getDocumentUrl(d);
+      if (this.conversionSettings.excludes.has(documentUrl)) {
+        return false;
+      }
+      // Filter out any external documents
+      const packageName =
+          this.urlHandler.getOriginalPackageNameForUrl(documentUrl);
+      return packageName === this.packageName;
+    });
+  }
+
+  /**
+   * Return the results of the package scan.
+   */
+  getResults(): PackageScanResult {
+    return {
+      files: this.results,
+      exports: this.namespacedExports,
+    };
+  }
+
+  /**
+   * Scan a document and any of its dependency packages. The scan document
+   * format (JS Module or HTML Document) is determined by whether the file is
+   * included in conversionSettings.includes.
+   */
+  private scanDocument(document: Document, forceJs = false) {
+    console.assert(
+        document.kinds.has('html-document'),
+        `scanDocument() must be called with an HTML document, but got ${
+            document.kinds}`);
+    if (!this._shouldScanDocument(document)) {
+      return;
+    }
+
+    const documentUrl = this.urlHandler.getDocumentUrl(document);
+    const documentConverter = new DocumentConverter(
+        document, this.urlHandler, this.conversionSettings);
+    let scanResult: JsModuleScanResult|HtmlDocumentScanResult|
+        DeleteFileScanResult;
+    try {
+      scanResult =
+          (forceJs || this.conversionSettings.includes.has(documentUrl)) ?
+          documentConverter.scanJsModule() :
+          documentConverter.scanTopLevelHtmlDocument();
+    } catch (e) {
+      console.error(`Error in ${document.url}`, e);
+      return;
+    }
+    this.results.set(scanResult.originalUrl, scanResult);
+    this._scanDependencies(document);
+    if (scanResult.type === 'js-module') {
+      for (const expr of scanResult.exportMigrationRecords) {
+        if (!this.namespacedExports.has(expr.oldNamespacedName)) {
+          this.namespacedExports.set(
+              expr.oldNamespacedName,
+              new JsExport(scanResult.convertedUrl, expr.es6ExportName));
+        }
+      }
+    }
+  }
+
+  /**
+   * Check if a document is explicitly excluded or has already been scanned
+   * to decide if it should be skipped.
+   */
+  private _shouldScanDocument(document: Document): boolean {
```

Comment:
Let's be consistent on private names. I disagree with the Google TS style guide, since `private` is stripped away in compilation, but 🤷‍♂️

Reply:
yea, I have mixed feelings here as well. Sticking without the leading `_` for now so that our codebase is consistent.